File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = list(set(nums))
        longest = 0
        for n in nums:
            if (n-1) not in nums:
                len = 0
                while (n + len) in nums:
                    len +=1
                longest = max(longest, len)
        return longest

File: Data Structures & Algorithms/trapping-rain-water/submission-4.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        let total = 0;

        let l = 0,
            r = height.length - 1;

        let maxL = height[l],
            maxR = height[r];

        while(l<r){
            if(maxL < maxR){
                // at any point the smaller side is the limiting factor that's why larger side is ignored
                l++;
                maxL = Math.max(maxL, height[l]);
                total += maxL - height[l];
            }else{
                r--;
                maxR = Math.max(maxR, height[r]);
                total += maxR - height[r];
            }

        }
        return total;
    }
}